Plot two UF fr tho

JsonHrdwick opened the pull request:
Allows user to plot two userfunctions against each other. Adds a new tab to analysis manager tabpane.

Issue #309

JsonHrdwick commented:
First pass commit contains the basics of getting this feature up and running. It has the logic to plot the UFs against each other without errors but there is more work that is needed for removing unused code from the altered OGplots class as well as ensuring mouse functionality and axis scaling is accurate.

noahmclean commented:
This sounds like a great start! 

To ensure we're on the same page, we'll want to plot isotope ratios and/or user functions against one another in this plot. So, one isotope ratio vs. another, one user function vs. another, or a user function vs. an isotope ratio. The user should be able to select any time series to plot against any other time series.

Jim asked me a while ago to figure out how to illustrate and implement data rejection on a 2D plot. How about this? 
    - Data whose x- and y-variables are both included get a color ramp as Jim describes, with color based on time. However, let's build on the default color palette from Tripoli and use circular markers with a "viridis" color palette. You can copy the implementation [here](https://waldyrious.net/viridis-palette-generator/) or other open source places if you're unfamiliar. So, instead of blue points being included as in the 1D time series, there will be a ramp from yellow through green and blue to purple, all relatively cool colors. The markers will likely need a thin black marker edge to show up on a white or light gray background.
    - Data that are rejected in both the x- and y- variable should get red square markers. The contrast between the red and viridis ramp should be ok, the red is consistent with the 1D time series, and the square marker should help those with color vision deficiency. 
    - Data that are rejected in the x-variable but not the y-variable should get a red downward-pointing triangle ▼ marker. Data that are rejected in the y-variable but not the x-variable should get a red left-pointing triangle ◀ marker. The red signals data rejection, and the triangle points toward the axis whose data is rejected.

Finally, it would be fun to scale the marker size by a third user-chosen parameter. So, markers would be colored by time and scaled by, e.g., one of the intensity user functions. This would likely require two legends -- a color bar legend and a size legend.

bowring commented:
rejection icons working well
let's turn off all sculpting actions and concentrate on zoom only
great work

JsonHrdwick commented:
Bunch of UI tweaks
  - Legends
  - Axis labels
  - Sizing
  - Text fonts
 
Implemented an optional intensity/scaling UF option (Hope that is how it should work)
Fixed mouse-wheel zooming

Needed:
Parameter for scaling factor/size range
Remove/implement unused functions
Clean up old code/comments

bowring commented:
@JsonHrdwick - Looking great!
1. consider possibly that zoom might need to reset min/max  x to provide margins for large dots?
2. sculpting a point that is rejected in both userfunctions correctly produces square, but on reset all data, only one userfunction is reset, leaving a triangle
3. if i plot anytwo and then plot userfunctions without closing anytwo, anytwo becomes unavailable
4. we will need a way to choose more than one anytwo plots and save them as part of the method.  these plots should be added to the grid of userfunctions as well
5. @noahmclean - how to handle the case where the intensity userfunction has rejects?  ignore them?
6. @bowring available for chat anytime

JsonHrdwick commented:
> if i plot anytwo and then plot userfunctions without closing anytwo, anytwo becomes unavailable

I wasn't able to re-produce this. On my end, plot2 was still usable but would just add another plot to the existing window. I made it mirror plot1 and reopen to be more uniform with the rest of the application.

bowring commented:
Great work
Enjoy thanksgiving!

Jim Bowring
Associate Professor Emeritus, Computer Science, College of Charleston
Principal Investigator, www.CIRDLES.org, www.github.com/cirdles
GitHub: https://github.com/bowring

R. Buckminster Fuller (1972):
If humanity is to survive aboard our planet, it must become universally
literate and preoccupied with inherently cooperative Comprehensive
Anticipatory Design Science in which every human is concerned with
accomplishing the comfortably sustainable well-faring of all other humans.


On Sun, Nov 23, 2025 at 3:01 PM Jason Hardwick ***@***.***>
wrote:

> *JsonHrdwick* left a comment (CIRDLES/Tripoli#338)
> <https://github.com/CIRDLES/Tripoli/pull/338#issuecomment-3568286046>
>
> Moving forward with ignoring intensity rejects.
> Decided against moving the plot2 in the main window to the end of the
> list. This would require much more work and on further thought it makes
> sense to keep it as is given the plot2 will always render directly after
> its x-axis value ***@***.*** <https://github.com/bowring>, @noahmclean
> <https://github.com/noahmclean>)
>
> Removed several unused features UI
> Fixed responsiveness of Plot2 buttons
> Added parameters to dot size in plot to (system default 5/20 min/max)
>
> Needed still:
> Making compact plot2 more readable
> Add intensity label to plot2 window
> handle log scale in plot2
> check all remaining buttons for unused functionality
>
> Sorry, gonna have to be a post thanksgiving release
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/CIRDLES/Tripoli/pull/338#issuecomment-3568286046>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAIK2YYZMQI6YMGKAT6DITL36IHC5AVCNFSM6AAAAACLS2UM5CVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTKNRYGI4DMMBUGY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

JsonHrdwick commented:
Good news everybody!

Think we're ready to go.
I've corrected any issues that I could find. This last burst of commits included a lot of UI tweaks:
- Removed unused buttons from Plot2 window.
- Removed functionality that is not need for Plot2
- Migrated the legends the left side so the layout better matches PlotOG (See Img1)
- Fix x-axis label to slide left when view gets too small
- Added selected function to intensity scale label
- Extra just for my Tripoli fans: Double-click on a saved Plot2 selection and it will pre-fill the dropdowns

<img width="1252" height="757" alt="07_T-1351_816" src="https://github.com/user-attachments/assets/388beff0-5578-4eab-9eee-85966f012639" />

<img width="1223" height="693" alt="07_T-1356_817" src="https://github.com/user-attachments/assets/d40258d1-c72e-4b75-b6c4-3f809fcbc6f7" />
